### PR TITLE
Clean Up Codebase

### PR DESCRIPTION
**Pull request by rasjonell**
- **Project**
  - Add shared `gitignore`
  - Refactor modules according to style-guide
- **Generator**
  - Bump version
- **Frontend**
  - Disable graph pan and zoom
  - Create modular Data/Util Layers for Modals/Dialogs/Actions

### CODE REVIEW

**Comment by rasjonell** [comment]
[Clean Up Documentation](https://trello.com/c/VU1mqR7W/6-clean-up-documentation)

**Comment by rasjonell** [comment]
[Clean Up The Codebase](https://trello.com/c/tDbpR8dM/12-clean-up-the-codebase)

File: frontend/src/components/Actions/data.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    name: 'Create A Collection',
+  },
+  {
+    type: 'node',
+    Icon: NodeIcon,
+    name: 'Create A Node',
+  },
+  {
+    type: 'link',
+    name: 'Create A Link',
```
Comment (rasjonell) [inline frontend/src/components/Actions/data.js:48]: I should reorder this

File: frontend/src/components/Modals/data.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    <span>
+      Your schema must have only one root-level field
+      <ul>
+        <li>the collection name</li>
+      </ul>
+    </span>
+  ),
+  invalidFields: (
+    <span>
+      Your collection fields must have only these fields:
```
Comment (rasjonell) [inline frontend/src/components/Modals/data.js:61]: I should think of a better wording

File: frontend/src/components/Dialogs/Link.js [hunk 3]
```diff
@@ -12,14 +12,21 @@ import DialogContent from '@material-ui/core/DialogContent';
...
-export const TYPE = 'link';
+/* Constants */
+
+const DEFAULT_STATE = { edge: '', _from: '', _to: '' };
+
+/* Exports */
 
-const defaultState = { edge: '', _from: '', _to: '' };
+export default Dialog;
+export const TYPE = 'link';
```
Comment (adsellor) [inline frontend/src/components/Dialogs/Link.js:27]: The notion of exporting a `TYPE` variable, is a bit confusing. It makes sense in the context, but if it's going to be exported and used outside, i think it is better to be renamed to `DIALOG_TYPE`.

**Pull request merged**